nowplaying/daemon.py

```diff
@@ -0,0 +1,85 @@
+import logging
+import os
+import signal
+import sys
+import time
+
+from input import observer as inputObservers
+from input.handler import InputHandler
+from track import observer as trackObservers
+from track.handler import TrackEventHandler
+
+logger = logging.getLogger("now-playing")
+
+
+class NowPlayingDaemon:
+    """The daemon part of the nowplaying app runs at all time and coordinates the input and output."""
+
+    def __init__(self, options):
+        self.options = options
+
+    def main(self):
+        logger.info("Starting up now-playing daemon")
+
+        try:
+            self.register_signal_handlers()
+
+            input_handler = self.get_input_handler()
+        except Exception as e:
+            logger.exception("Error: %s", e)
+            sys.exit(-1)
+
+        while True:
+            try:
+                input_handler.update()
+            except Exception as e:
+                logger.exception("Error: %s", e)
+
+            time.sleep(self.options.sleepSeconds)
+
+    def register_signal_handlers(self):
+        logger.debug("Registering signal handler")
+        signal.signal(signal.SIGINT, self.signal_handler)
+        # signal.signal(signal.SIGKIL, self.signal_handler)
+
+    def signal_handler(self, signum, frame):
+        logger.debug("Signal %i caught" % signum)
+
+        if signum == signal.SIGINT or signum == signal.SIGKILL:
+            logger.info("Signal %i caught, terminating." % signum)
+            sys.exit(os.EX_OK)
+
+    def get_track_handler(self):
```

Comment:
This method was unified into one. It used to be called `get_klangbecken_track_handler` and `get_nonklangbecken_track_handler`. During refactoring I realized that they both had the same content.